### PR TITLE
Mnist update

### PR DESCRIPTION
**Pull request by surgan12**
Changing the architecture of the Mnist , simplifying it and increasing the accuracy.
Also added the functionalty of saving the model as an argument.

### CODE REVIEW

File: mnist/main.py [hunk 1]
```diff
@@ -6,24 +6,27 @@
...
+        x = F.relu(self.conv2(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = x.view(-1, 4*4*50)
         x = F.relu(self.fc1(x))
-        x = F.dropout(x, training=self.training)
         x = self.fc2(x)
         return F.log_softmax(x, dim=1)
-
+    
+    def name(self):
```
Comment (soumith) [inline mnist/main.py]: this function should be removed

File: mnist/main.py [hunk 2]
```diff
@@ -100,10 +107,14 @@ def main():
...
 
-    for epoch in range(1, args.epochs + 1):
+    for epoch in range(1, args.epochs+1):
         train(args, model, device, train_loader, optimizer, epoch)
         test(args, model, device, test_loader)
 
+    if (args.save_model):
+        PATH="mnist_cnn.pt"
+        torch.save(model.state_dict(), PATH)
+        print("model saved as mnist_cnn.pt in the current working directory\n")
```
Comment (soumith) [inline mnist/main.py]: print is unnecessary

File: mnist/main.py [hunk 3]
```diff
@@ -100,10 +107,14 @@ def main():
     model = Net().to(device)
     optimizer = optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
 
-    for epoch in range(1, args.epochs + 1):
+    for epoch in range(1, args.epochs+1):
         train(args, model, device, train_loader, optimizer, epoch)
         test(args, model, device, test_loader)
 
+    if (args.save_model):
+        PATH="mnist_cnn.pt"
```
Comment (soumith) [inline mnist/main.py]: additional PATH is unncessary, do `torch.save(model.state_dict(), "mnist_cnn.pt")`

File: mnist/main.py [hunk 4]
```diff
@@ -100,10 +107,14 @@ def main():
     model = Net().to(device)
     optimizer = optim.SGD(model.parameters(), lr=args.lr, momentum=args.momentum)
 
-    for epoch in range(1, args.epochs + 1):
+    for epoch in range(1, args.epochs+1):
```
Comment (soumith) [inline mnist/main.py]: `args.epochs + 1` is cleaner. Revert this change

**Comment by surgan12** [comment]
@soumith  i have made the changes as asked by you.
Kindly look into it .
Thanks

**Comment by soumith** [comment]
thank you @surgan12 !

**Comment by surgan12** [comment]
@soumith  my pleasure always !
